controller/src/test/java/io/pravega/controller/server/bucket/PravegaTablesStoreBucketServiceTest.java

```diff
@@ -281,5 +281,44 @@ public void testNodeCacheFailure() throws Exception {
         AssertExtensions.assertThrows(IllegalStateException.class, () -> bucketManager.awaitRunning());
     }
 
+    @Test(timeout = 30000)
+    public void testSplitBrainScenrio() throws Exception {
+        addEntryToZkCluster(controller);
+
+        Host controller1 = new Host(UUID.randomUUID().toString(), 9090, null);
+        BucketManager bucketManager1 = getBucketManager(controller1.getHostId(), "mypath");
+        addEntryToZkCluster(controller1);
+
+        Host controller2 = new Host(UUID.randomUUID().toString(), 9090, null);
+        BucketManager bucketManager2 = getBucketManager(controller2.getHostId(), BucketStore.ServiceType.WatermarkingService.getName());
+        addEntryToZkCluster(controller2);
+
+        bucketManager1.startAsync().awaitRunning();
+        bucketManager2.startAsync().awaitRunning();
 
+        //Check we have two leaders for same cluster.
+        assertEventuallyEquals(true, () -> ((ZooKeeperBucketManager) bucketManager1).isLeader(), 1000);
+        assertEventuallyEquals(true, () -> ((ZooKeeperBucketManager) bucketManager2).isLeader() ||
+                ((ZooKeeperBucketManager) watermarkingService).isLeader(), 1000);
+
+        //Check buckets are evenly distributed
+        assertEventuallyEquals(1, () -> bucketManager1.getBucketServices().size(), 10000);
+        assertEquals(1, bucketManager2.getBucketServices().size());
+        assertEquals(1, watermarkingService.getBucketServices().size());
+
+        //Stop bucket manager
+        bucketManager1.stopAsync().awaitTerminated();
+        bucketManager2.stopAsync().awaitTerminated();
```

Comment:
I am not clear on what is really being asserted here. If two hosts can become leader the concern is that they will work inconsistently or both take on the same job. This test merely seems to assert that split brain can happen.

Reply:
In this test case, try to simulate split brain situation.

- At line no 300and 301, asserted that there are two leaders exits.
- At line no 305-307, i asserted that even though split brain occurs but still buckets are evenly distributed. In this test case, have total 3 buckets for watermarking. All three bucket manager instances is having equal no of buckets.

I have added more assert to verify every instance is having identical buckets.